### PR TITLE
Fix commit message validation for patchback

### PR DESCRIPTION
**Pull request by fao89**
[noissue]

### CODE REVIEW

**Comment by fao89** [comment]
Tested with:
- https://github.com/pulp/pulp_ansible/pull/786
- https://github.com/pulp/pulp_ansible/pull/787

File: templates/github/.ci/scripts/validate_commit_message.py.j2 [hunk 1]
```diff
@@ -28,7 +28,8 @@ def __check_status(issue):
     gi = repo.get_issue(int(issue))
     if gi.pull_request:
         sys.exit(f"Error: issue #{issue} is a pull request.")
-    if gi.closed_at:
+    PR_BODY = os.environ.get("GITHUB_PULL_REQUEST_BODY", "")
+    if gi.closed_at and "backport" not in PR_BODY:
```
Comment (mdellweg) [inline templates/github/.ci/scripts/validate_commit_message.py.j2:32]: Do we maybe want to search for "cherry picked from commit" in the commit message instead?

Comment (mdellweg) [reply templates/github/.ci/scripts/validate_commit_message.py.j2:32]: Maybe even check, that a cherrypick does not contain a feature?

Comment (fao89) [reply templates/github/.ci/scripts/validate_commit_message.py.j2]: done!
Now patchback PR fails for the right reason:
https://github.com/pulp/pulp_ansible/runs/4716629728?check_suite_focus=true

File: templates/github/.ci/scripts/validate_commit_message.py.j2 [hunk 2]
```diff
@@ -28,7 +29,10 @@ def __check_status(issue):
     gi = repo.get_issue(int(issue))
     if gi.pull_request:
         sys.exit(f"Error: issue #{issue} is a pull request.")
-    if gi.closed_at:
+    if gi.closed_at and "cherry picked from commit" not in message:
+        warnings.warn(
+            "When backporting, make sure to have 'cherry picked from commit' in the commit message."
```
Comment (mdellweg) [inline templates/github/.ci/scripts/validate_commit_message.py.j2:34]: "... use the -x flag to ..."

**Review by mdellweg — APPROVED** [review]

File: templates/github/.ci/scripts/validate_commit_message.py.j2 (hunk 2 above)
Comment (mdellweg) [reply templates/github/.ci/scripts/validate_commit_message.py.j2:34]: ^ optional

**Comment by mdellweg** [comment]
This is great!

**Pull request merged**